core/src/software/aws/toolkits/core/credentials/ToolkitCredentialsProvider.kt

```diff
@@ -35,19 +35,19 @@ abstract class ToolkitCredentialsProvider : AwsCredentialsProvider {
 
     override fun toString(): String = "${this::class.simpleName}(id='$id')"
 
+    /**
+     * Returns true or throws an Exception
+     */
+    @Throws(Exception::class)
     open fun isValid(sdkHttpClient: SdkHttpClient): Boolean {
```

Comment:
Can we rename it to `validateIdentity` and return `void`. A method that always returns `true` or throw exception seems weird to me.

Reply:
renamed to `isValidOrThrow()` to clarify the contract, since we still want the boolean return